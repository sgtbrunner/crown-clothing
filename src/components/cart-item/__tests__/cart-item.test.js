import React from 'react';
import { render, screen } from '@testing-library/react';

import CartItem from '../cart-item.component';

const props = {
  id: 1,
  price: 10,
  name: 'Item',
  quantity: 2,
  imageUrl: 'https://www.myimage.com/item',
};

const mountComponent = () => render(<CartItem item={props} />);

describe('CartItem component', () => {
  it('should match component snapshot', () => {
    const { container } = mountComponent();

    expect(container).toMatchSnapshot();
  });

  it('should render the item image', () => {
    mountComponent();

    const image = screen.getByAltText('item');
    expect(image).toBeInTheDocument();
    expect(image).toHaveProperty('src', props.imageUrl);
  });

  it('should render the item details', () => {
    mountComponent();

    const itemName = screen.getByText(props.name);
    expect(itemName).toBeInTheDocument();

    const itemQuantPrice = screen.getByText(
      `${props.quantity} x $${props.price}`
    );
    expect(itemQuantPrice).toBeInTheDocument();
  });
});
