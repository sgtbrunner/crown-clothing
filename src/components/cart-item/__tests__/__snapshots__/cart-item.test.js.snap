// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem component should match component snapshot 1`] = `
<div>
  <div
    class="sc-bdnxRM kZjATh"
  >
    <img
      alt="item"
      class="sc-gtsrHT dSkxJL"
      src="https://www.myimage.com/item"
    />
    <div
      class="sc-dlnjwi kRPREb"
    >
      <span>
        Item
      </span>
      <span>
        2
         x $
        10
      </span>
    </div>
  </div>
</div>
`;
